File: vvvv45/addonpack/src/nodes/plugins/Layer/DrawText/SlimDXUtils.cs
using VVVV.Utils.VMath;
using SlimDX;

namespace VVVV.Utils
{
	public sealed class VSlimDXUtils
	{
		public static SlimDX.Matrix Matrix4x4ToSlimDXMatrix(Matrix4x4 Input)
		{
			SlimDX.Matrix Result = new Matrix();
			
			Result.M11 = (float) Input.m11;
			Result.M12 = (float) Input.m12;
			Result.M13 = (float) Input.m13;
			Result.M14 = (float) Input.m14;
			
			Result.M21 = (float) Input.m21;
			Result.M22 = (float) Input.m22;
			Result.M23 = (float) Input.m23;
			Result.M24 = (float) Input.m24;
			
			Result.M31 = (float) Input.m31;
			Result.M32 = (float) Input.m32;
			Result.M33 = (float) Input.m33;
			Result.M34 = (float) Input.m34;
			
			Result.M41 = (float) Input.m41;
			Result.M42 = (float) Input.m42;
			Result.M43 = (float) Input.m43;
			Result.M44 = (float) Input.m44;
			
			return Result;
		}
	}
}

File: vvvv45/src/core/SharedSources/SlimDX/SlimDXUtils.fs
module VVVV.SlimDXUtils

open SlimDX
open VVVV.Utils.VMath

let inline Mat2SDXMat (m:Matrix4x4) = new Matrix(M11 = float32 m.m11, M12 = float32 m.m12, M13 = float32 m.m13, M14 = float32 m.m14,
                                                 M21 = float32 m.m21, M22 = float32 m.m22, M23 = float32 m.m23, M24 = float32 m.m24,
                                                 M31 = float32 m.m31, M32 = float32 m.m32, M33 = float32 m.m33, M34 = float32 m.m34,
                                                 M41 = float32 m.m41, M42 = float32 m.m42, M43 = float32 m.m43, M44 = float32 m.m44)

